### PR TITLE
[4.2] [UX] Reset hidden postinstallation messages

### PR DESCRIPTION
**Pull request by magnussinger**
Pull Request for Issue #37399

### Summary of Changes
In the post-installation messages, you can now reset hidden messages before you have to hide all the other messages

### Testing Instructions
There's an extra button in the toolbar, hide one message and press the reset button, and the message will be visible again

### Actual result BEFORE applying this Pull Request
You had to hide all messages before making them visible again

### Expected result AFTER applying this Pull Request
You can show hidden messages again before hiding all others before

### Documentation Changes Required
None

### CODE REVIEW

**Comment by brianteeman** [comment]
The reset button should not be displayed (or it should be disabled) if there are no hidden messages.

(I'm not convinced this button is a suitable solution to the issue. As I write in the issue there are several ways to achieve the aim. It all depends on what is the motivation/reason for the change.

**Comment by chmst** [comment]
Thank you for your contribution, @MagnusSinger! 

This indeed solves the acute problem, that a hidden message cannot be reactivated. As you can see in the issue comments, this could be only an interim solution.

**Comment by magnussinger** [comment]
Yes I'll fix that it only shows the button when messages are hidden later, but I think the solution with the button on top is a bit cleaner than minimizing hidden messages, but I'm open for suggestions

**Comment by magnussinger** [comment]
@brianteeman I just updated my PR and it shows the button only when there are hidden messages now

File: administrator/components/com_postinstall/src/Model/MessagesModel.php [hunk 1]
```diff
@@ -189,6 +189,23 @@ public function getItems()
 		return $result;
 	}
 
+	/**
+	 * Get hidden messages
+	 *
+	 * @return array  the hidden messages
+	 *
+	 * @since 4.2
```
Comment (Quy) [inline administrator/components/com_postinstall/src/Model/MessagesModel.php]:
```suggestion
	 * @since  __DEPLOY_VERSION__
```

**Comment by Quy** [comment]
Please fix and use the refresh icon instead of plus icon.

security-token

**Comment by magnussinger** [comment]
@Quy I updated the button

**Comment by Quy** [comment]
Please also fix the security token warning. Thanks.

File: administrator/components/com_postinstall/src/View/Messages/HtmlView.php [hunk 2]
```diff
@@ -82,7 +82,7 @@ private function toolbar()
 
 			if (!empty($this->hidden))
 			{
-				$toolbar->standardButton('new', 'COM_POSTINSTALL_EMPTYSTATE_BUTTON_ADD', 'message.reset');
+				$toolbar->standardButton('', 'COM_POSTINSTALL_EMPTYSTATE_BUTTON_ADD', 'message.reset')->icon('icon-refresh');
```
Comment (Quy) [inline administrator/components/com_postinstall/src/View/Messages/HtmlView.php]: Please format at follows:

https://github.com/MagnusSinger/joomla-cms/blob/0d1d8632dd0cdaf73e33206b937933da5a0aae96/administrator/components/com_menus/src/View/Items/HtmlView.php#L398-L400

**Comment by magnussinger** [comment]
@Quy I'm sorry but how can I see warnings in the backend? I somehow can't see the warning message

**Comment by Quy** [comment]
Click the Reset button in the toolbar.

File: administrator/components/com_postinstall/src/View/Messages/HtmlView.php [hunk 3]
```diff
@@ -78,6 +79,11 @@ private function toolbar()
 		if (!empty($this->items))
 		{
 			$toolbar->unpublish('message.hideAll', 'COM_POSTINSTALL_HIDE_ALL_MESSAGES');
+
+			if (!empty($this->hidden))
+			{
+				$toolbar->standardButton('refresh', 'COM_POSTINSTALL_EMPTYSTATE_BUTTON_ADD', 'message.reset')->icon('icon-refresh');
```
Comment (Quy) [inline administrator/components/com_postinstall/src/View/Messages/HtmlView.php]:
```suggestion
				$toolbar->standardButton('refresh', 'COM_POSTINSTALL_EMPTYSTATE_BUTTON_ADD', 'message.reset');
```

**Comment by magnussinger** [comment]
When I click on the reset button, there is no warning message, I just re-tried it

**Comment by Quy** [comment]
Let me try again with the prebuilt package and report back later.

**Comment by magnussinger** [comment]
Thanks

**Comment by Quy** [comment]
Closing in favor of #38064. Thank you.
